examples/src/keyring/aws_kms/act_like_aws_kms_master_key_provider.py

```diff
@@ -0,0 +1,88 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Before there were keyrings, there were master key providers.
+Master key providers were the original configuration structure
+that we provided for defining how you want to protect your data keys.
+
+The AWS KMS master key provider was the tool that we provided for interacting with AWS KMS.
+Like the AWS KMS keyring,
+the AWS KMS master key provider encrypts with all CMKs that you identify,
+but unlike the AWS KMS keyring,
+the AWS KMS master key provider always attempts to decrypt
+*any* data keys that were encrypted under an AWS KMS CMK.
+We have found that separating these two behaviors
+makes it more clear what behavior to expect,
+so that is what we did with the AWS KMS keyring and the AWS KMS discovery keyring.
+However, as you migrate away from master key providers to keyrings,
+you might need to replicate the behavior of the AWS KMS master key provider.
+
+This example shows how to configure a keyring that behaves like an AWS KMS master key provider.
+
+For more examples of how to use the AWS KMS keyring,
+see the ``keyring/aws_kms`` directory.
+"""
+import aws_encryption_sdk
+from aws_encryption_sdk.key_providers.kms import KMSMasterKeyProvider
+from aws_encryption_sdk.keyrings.aws_kms import AwsKmsKeyring
+from aws_encryption_sdk.keyrings.multi import MultiKeyring
+
+
+def run(aws_kms_cmk, source_plaintext):
+    # type: (str, bytes) -> None
+    """Demonstrate how to create a keyring that behaves like an AWS KMS master key provider.
+
+    :param str aws_kms_cmk: The ARN of an AWS KMS CMK that protects data keys
```

Comment:
If we're using the same keyring for encryption and decryption, this must be a [key ARN](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id-key-ARN).

Reply:
With an AWS KMS master key or an AWS KMS keyring, yes, that is correct. However, with an AWS KMS master key provider or an AWS KMS discovery keyring, which is what we are demonstrating here, it doesn't matter.

Reply:
I still think it's best to show key ARNs unless there's a reason to do otherwise, but I won't press the issue.

Reply:
The main reasons for pulling the ARNs in via parameters rather than showing them directly are to avoid anyone copy-pasting accidentally using our test CMKs anywhere and so that our test runners can be more flexible if we want to change the specific CMKs that they use.